src/provider/application.js

```diff
@@ -231,6 +233,11 @@ class Application extends EventEmitter {
   emitError(error) {
     this.emit('xfc.error', error);
   }
+
+  unload() {
+    this.JSONRPC.notification('unload');
+    this.trigger('xfc.unload', { url: document.activeElement.href });
```

Comment:
Do you know if we need url? I ask because this code has a few situations where it won't behave as expected..

* unloading from submitting a form
* unloading from clicking on a non-anchor elmenet triggering javascript redirect
* unloading from javascript just navigating away.

I'm concerned that it will throw an error if there isn't an activeElement and that the code listening will always expect a url when I think there are lots of scenarios there won't be one.

Reply:
We need it if we are going to use this as the redirect dictionary as well. It's how we get the portal path before loading the new pagelet.

Reply:
Talked to @mhemesath  about this. Let's keep the redirect dictionary in place for now and make no changes to it in the SDK/Healthelastic-React. That way we can remove the URL here + redirect dictionary will be ready to refactor in Q2.

Reply:
@mhemesath I think we are fine as-is. I tested in Chrome,FF,Safari and activeElement.href will at worst be undefined.

Reply:
Sounds good - I'll take the URL out here and keep the old redirect in react

Reply:
@slor  if active element is undefined then this code will throw an error as is. Even putting a conditional check in place I still recommend removing URL as its presence is unreliable

Reply:
I agree with @mhemesath. We should not trigger this event with URL every time before page unloads, since it'll error out in a couple scenarios (A very simple example is refreshing the current page). I think we're doing here is not actually catching unload event. Instead, we're looking for click event that triggers page redirection. Therefore, first I feel the event name should be changed to something like "xfc.redirect" and we'll check if activeElement has href attribute. If we do want to listen for unload and redirect, we might want to split them into two separate events.

Reply:
` I think what we're doing here is not actually catching unload event. Instead`  @kafkahw  look at line 121, this code is handled on the `beforeunload` event.. that means anything that causes the browser to navigate away.

Reply:
@mhemesath OK, I am fine removing it since other scripts in the same frame can get the activeElement's href without XFC supplying it.